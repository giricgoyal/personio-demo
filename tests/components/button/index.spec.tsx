import { fireEvent } from '@testing-library/react'
import React from 'react'
import Button from 'src/components/button'
import { render } from 'tests/test-utils'

describe('component/button', () => {
    let container, getByTestId, onClickHandler

    beforeEach(() => {
        onClickHandler = jest.fn()
        ;({ container, getByTestId } = render(<Button onClick={onClickHandler}>Label</Button>))
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('Click event', () => {
        beforeEach(() => {
            fireEvent.click(getByTestId('button'))
        })

        test('should call the onClickHandler', () => {
            expect(onClickHandler).toHaveBeenCalledTimes(1)
        })
    })
})
