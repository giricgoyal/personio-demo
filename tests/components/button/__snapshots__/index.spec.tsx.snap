// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/button should render and match snapshot 1`] = `
<div>
  <button
    class="button button--primary"
    data-testid="button"
  >
    Label
  </button>
</div>
`;
